modules/nf-scil/reconst/frf/tests/main.nf.test

```diff
@@ -0,0 +1,249 @@
+                        file("\${test_data_directory}/dwi/dwi.bval", checkIfExists: true),
+                        file("\${test_data_directory}/dwi/dwi.bvec", checkIfExists: true),
+                        [],
+                        [],
+                        [],
+                        []
+                    ]
+                }
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out).match() }
+            )
+        }
+
+    }
+
+    test("reconst - frf_msmt") {
+
+        config "./nextflow_msmt.config"
+
+        when {
+            process {
+                """
+                ch_split_test_data = LOAD_DATA.out.test_data_directory
+                    .branch{
+                        ssst: it.simpleName == "heavy"
+                        msmt: it.simpleName == "commit_amico"
+                    }
+                input[0] = ch_split_test_data.msmt.map{
+                    test_data_directory -> [
+                        [ id:'test', single_end:false ], // meta map
+                        file("\${test_data_directory}/dwi.nii.gz", checkIfExists: true),
+                        file("\${test_data_directory}/dwi.bval", checkIfExists: true),
+                        file("\${test_data_directory}/dwi.bvec", checkIfExists: true),
+                        file("\${test_data_directory}/mask.nii.gz", checkIfExists: true),
+                        [],
+                        [],
+                        []
+                    ]
+                }
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out).match() }
+            )
+        }
+
+    }
+
+    test("reconst - frf_msmt_set_frf") {
+
+        config "./nextflow_msmt_set_frf.config"
+
+        when {
+            process {
+                """
+                ch_split_test_data = LOAD_DATA.out.test_data_directory
+                    .branch{
+                        ssst: it.simpleName == "heavy"
+                        msmt: it.simpleName == "commit_amico"
+                    }
+                input[0] = ch_split_test_data.msmt.map{
+                    test_data_directory -> [
+                        [ id:'test', single_end:false ], // meta map
+                        file("\${test_data_directory}/dwi.nii.gz", checkIfExists: true),
+                        file("\${test_data_directory}/dwi.bval", checkIfExists: true),
+                        file("\${test_data_directory}/dwi.bvec", checkIfExists: true),
+                        file("\${test_data_directory}/mask.nii.gz", checkIfExists: true),
+                        [],
+                        [],
+                        []
+                    ]
+                }
+                """
+            }
+        }
```

Comment:
It would be worth adding a test validating the behavior when `wm_mask`, `gm_mask`, and `csf_mask` are provided.

Reply:
We would need to add these masks to the test data. Future work?

Reply:
Yep, I agree.